kubetest/kops.go

```diff
@@ -414,12 +414,19 @@ func (k kops) Up() error {
 	if k.adminAccess == "" {
 		var b bytes.Buffer
 		err := httpReadWithHeaders(externalIPURL, map[string]string{"Metadata-Flavor": "Google"}, &b)
-		if err != nil || net.ParseIP(strings.TrimSpace(b.String())) == nil {
+
+		count := 0
+		for count < 5 && err != nil && net.ParseIP(strings.TrimSpace(b.String())) == nil {
+			time.Sleep(2 * time.Second)
+			count++
 			b.Reset()
-			if err := httpRead("https://v4.ifconfig.co", &b); err != nil {
-				return err
-			}
+			err = httpRead("https://v4.ifconfig.co", &b)
```

Comment:
how about show fail log count in here?

Reply:
I don't think it's so important. If it fails the 5th time, the error will probably be the same and will be displayed later.
If it doesn't fail, no one will look for the counter.

Reply:
Sure . get it 
/LGTM

Reply:
Thanks @tanjunchen